packages/primitives/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,764 @@
+         * update the value to that item's value. Otherwise, we want to update
+         * the value to the previous value.
+         *
+         * If theres no previous value and we've typed a value that doesn't match
+         * an item, we want to clear the value.
+         */
+        if (activeItem) {
+          context.onValueChange(activeItem.value);
+        } else if (previousItem) {
+          context.onTextValueChange(previousItem.textValue);
+        } else {
+          context.onTextValueChange('');
+        }
+
+        context.onSearchValueChange('');
+      })}
+    />
+  );
+});
+
+/* -------------------------------------------------------------------------------------------------
+ * ComboboxIcon
+ * -----------------------------------------------------------------------------------------------*/
+
+type ComboboxIconElement = React.ElementRef<typeof Primitive.button>;
+type PrimitiveButtonProps = ComponentPropsWithoutRef<typeof Primitive.button>;
+interface IconProps extends PrimitiveButtonProps {}
+
+const ComboboxIcon = React.forwardRef<ComboboxIconElement, IconProps>((props, forwardedRef) => {
+  const { children, ...iconProps } = props;
+
+  const context = useComboboxContext(INPUT_NAME);
+
+  const isDisabled = context.disabled;
+
+  const handleOpen = () => {
+    if (!isDisabled) {
+      context.onOpenChange(true);
+      /**
+       * We should be focussing the trigger here not this button.
+       */
+      context.trigger?.focus();
+    }
+  };
+
+  return (
+    <button
+      aria-hidden
+      {...iconProps}
+      tabIndex={-1}
+      ref={forwardedRef} // Enable compatibility with native label or custom `Label` "click" for Safari:
+      onClick={composeEventHandlers(iconProps.onClick, () => {
+        // Whilst browsers generally have no issue focusing the trigger when clicking
+        // on a label, Safari seems to struggle with the fact that there's no `onClick`.
+        // We force `focus` in this case. Note: this doesn't create any other side-effect
+        // because we are preventing default in `onPointerDown` so effectively
+        // this only runs for a label "click"
+        context.trigger?.focus();
+      })}
+      onPointerDown={composeEventHandlers(iconProps.onPointerDown, (event) => {
+        // only call handler if it's the left button (mousedown gets triggered by all mouse buttons)
+        // but not when the control key is pressed (avoiding MacOS right click)
+        if (event.button === 0 && event.ctrlKey === false) {
+          handleOpen();
+          // prevent trigger from stealing focus from the active item after opening.
+          event.preventDefault();
+        }
+      })}
+      onKeyDown={composeEventHandlers(iconProps.onKeyDown, (event) => {
+        if (OPEN_KEYS.includes(event.key)) {
+          handleOpen();
+          event.preventDefault();
+        }
+      })}
+    >
+      {children || '▼'}
```

Comment:
No I don't think we should, i think this should be as universal as possible. In the DS we'll replace it with our own icon because it's stylised to _our_ needs.